dune-project

```diff
@@ -60,3 +60,26 @@
     (>= "1.17")
     (< "1.18")))
   ("odoc" :with-doc)))
+
+(package
+ (name linol-eio)
+ (synopsis "LSP server library (with Eio for concurrency)")
+ (depends
+  ("yojson"
+   (>= "1.6"))
+  ("linol"
+   (= :version))
+  base-unix
+  ("eio_main"
```

Comment:
I don't think a library should depend on `eio_main`, only on `eio`, is that correct? The final executable you make using linol-eio will pick its backend (and might want to avoid eio_main because io_uring is not that secure currently)

Reply:
That's a good point. I think that eio_main dispatches to eio_posix or eio_linux (here is where it would use `io_uring`). The eio_main documentation says that `Eio_main.run` should not be called in library code, but for this PR it's only really being used to get the types of the environment for stdio. I'm not sure if perhaps eio_posix would be a better choice here, I'll take a look.